### PR TITLE
Missions

### PR DESCRIPTION
**Pull request by PeachesMLG**
Added Missions
(Nearly finished)

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/database/Island.java [hunk 1]
```diff
@@ -71,6 +74,16 @@ public Island(@NotNull String name, @NotNull Schematics.SchematicConfig schemati
...
+        this.experience = 0;
+    }
+
+    /**
+     * Gets the island's level
+     *
+     * @return The islands level
+     */
+    public int getLevel() {
+        return (int) Math.cbrt(experience + 1);
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/database/Island.java:86]: This is a pretty bad way to calculate the level IMO

File: src/main/java/com/iridium/iridiumskyblock/Reward.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.bukkit.entity.Player;
+
+import java.util.List;
+
+@NoArgsConstructor
+public class Reward {
+    public Item item;
+    public List<String> commands;
+    public int islandExperience;
+    public XSound sound;
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/Reward.java:16]: Why do you have a sound for every reward if you only use the one of the mission?

File: src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java [hunk 3]
```diff
@@ -378,6 +416,68 @@ private void recalculateIsland(@NotNull Island island, @NotNull List<ChunkSnapsh
...
     }
 
+    /**
+     * Increments a mission's data based on requirements
+     *
+     * @param island      The island
+     * @param missionData The mission data e.g. BREAK:COBBLESTONE
+     * @param increment   The amount we are incrementing by
+     */
+    public void incrementMission(@NotNull Island island, @NotNull String missionData, int increment) {
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java:426]: This method is a mess, split it up and add comments or something like that

**Review by dlsf — CHANGES_REQUESTED** [review]
I've seen worse things

File: src/main/java/com/iridium/iridiumskyblock/database/Island.java (hunk 1 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/database/Island.java:86]: There are multiple reasons for this.
First off, Math.cbrt() is very slow by a large margin even in comparison to other functions like Math.sqrt() which are already considered to be too slow: https://replit.com/@Dasunterstrich/InternationalSeagreenRom
Next, the graph for this function is simply not good for levels. The first three levels only need 99 XP, while level 10 already requires 1000, which in my opinion is too steep / a too big difference.

File: src/main/java/com/iridium/iridiumskyblock/Reward.java (hunk 2 above)
Comment (PeachesMLG) [reply src/main/java/com/iridium/iridiumskyblock/Reward.java:16]: Rewards havent been implemented yet, alot of people wanted commands as rewards so to do this ima have a /is rewards gui the sound is going to be the sound that plays where you collect the rewards from that GUI

File: src/main/java/com/iridium/iridiumskyblock/database/Island.java (hunk 1 above)
Comment (PeachesMLG) [reply src/main/java/com/iridium/iridiumskyblock/database/Island.java:86]: ye I didnt really know what to do for levels feel free to replace it with another function at a different time, but for now ima just leave it for testing purposes

Comment (PeachesMLG) [reply src/main/java/com/iridium/iridiumskyblock/database/Island.java:86]: Also its not that big of a deal according to your code cube rooting 1000000 times only takes 130ms, shouldnt be that big of a deal

**Pull request merged**